sauna/plugins/ext/supervisor.py

```diff
@@ -0,0 +1,166 @@
+from sauna.plugins import Plugin, PluginRegister
+import xmlrpc.client
+import http.client
+import socket
+
+my_plugin = PluginRegister('Supervisor')
+
+
+@my_plugin.plugin()
+class Supervisor(Plugin):
+
+    def __init__(self, config):
+        super().__init__(config)
+
+        serverurl = config.get('serverurl', 'unix:///var/run/supervisor.sock')
+        if serverurl.startswith('unix://'):
+            serverurl = serverurl.replace('unix://', '', 1)
+            # xmlrpc.client does not support Unix sockets, so we must provide
+            # a custom transport layer
+            transport = UnixStreamTransport(serverurl)
+            server = xmlrpc.client.ServerProxy('http://noop',
+                                               transport=transport)
+        else:
+            server = xmlrpc.client.ServerProxy(serverurl)
```

Comment:
When the server is not accessible sauna hangs. Could you specify a low timeout by default, like 5 seconds? You can reproduce with: `serverurl: http://user:123@8.8.8.8:10001/RPC2`